src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.spec.ts

```diff
@@ -35,6 +35,34 @@ describe('OsdListComponent', () => {
     }
   };
 
+  const getTableAction = (name) => component.tableActions.find((action) => action.name === name);
+
+  const setFakeSelection = () => {
+    // Default data and selection
+    const selection = [{ id: 1 }];
+    const data = [{ id: 1 }];
+
+    // Table data and selection
+    component.selection = new CdTableSelection();
+    component.selection.selected = selection;
+    component.selection.update();
+    component.osds = data;
+  };
+
+  const openActionModal = (actionName) => {
+    setFakeSelection();
+    getTableAction(actionName).click();
+  };
+
+  const mockSafeToDestroy = () => {
```

Comment:
Now that the creation of the spy has it's own function, it makes sense to move the comments to be part of the function, so that documentation is created appropriately.